doc/api/fs.md

```diff
@@ -283,6 +283,97 @@ synchronous use libuv's threadpool, which can have surprising and negative
 performance implications for some applications. See the
 [`UV_THREADPOOL_SIZE`][] documentation for more information.
 
+## Class: fs.DirectoryEntry
+<!-- YAML
+added: REPLACEME
+-->
+
+When [`fs.readdir()`][] or [`fs.readdirSync()`][] is called with the
+`withFileTypes` option set to `true`, the resulting array is filled with
+`fs.DirectoryEntry` objects, rather than strings or `Buffers`.
+
+Note that on some systems, getting file types at the same time as `readdir` is
+not supported. On those systems, only `dirent.isUnknown()` will return true, and
+all other `DirectoryEntry` methods will return false. To get file types on those
+systems, use [`fs.stat()`][].
```

Comment:
I’m wondering whether it might make sense to use `fs.stat()` in those cases in Node.js. The user wouldn’t be requesting `withFileTypes` if they aren’t going to use it, right? I know it sounds like a lot of overhead, but moving that overhead to userland doesn’t really buy us that much, does it?

Reply:
Good call I'll implement this when I have some time in the next week or so.

Reply:
The latest squash commit I pushed does this. PTAL.

Reply:
Does that mean that this note can be removed now? :)